mantle/kola/tests/misc/network.go

```diff
@@ -294,19 +305,196 @@ var (
               echo "ovs bridge already present"
           fi
 `
+
+	setupVethPairs = `#!/usr/bin/env bash
+          set -ex
+
+          create_veth_pair() {
+            veth_name=$1
+            host_side_mac_address=$2
+            container_side_cidr=$3
+
+            # create veth pair and assign a namespace to veth-container
+		    ip link add ${veth_name}-host type veth peer name ${veth_name}-container
+		    ip link set ${veth_name}-container netns container
+
+		    # assign an MAC address to the host-side veth end
+		    ip link set dev ${veth_name}-host address ${host_side_mac_address}
+
+		    # assign an IP address to container-side veth end and bring it up
+		    ip netns exec container ip address add ${container_side_cidr} dev ${veth_name}-container
+		    ip netns exec container ip link set ${veth_name}-container up
+          }
+          if [[ ! -f /boot/mac_addresses ]] ; then
+            echo "no mac address configuration file found .. skipping capture-veth-pairs"
+            exit 0
+          fi
+
+         echo "configure veth pairs"
+	 	 primary_mac=$(cat /boot/mac_addresses | awk -F= '/PRIMARY_MAC/ {print $2}')
+		 secondary_mac=$(cat /boot/mac_addresses | awk -F= '/SECONDARY_MAC/ {print $2}')
+
+		 # create a network namespace
+		 ip netns add container
+
+		 create_veth_pair veth1 $primary_mac 192.168.0.59
+		 create_veth_pair veth2 $secondary_mac 192.168.0.60
+`
 )
 
+func NetworkBondWithDhcp(c cluster.TestCluster) {
+	primaryMac := "52:55:00:d1:56:00"
+	secondaryMac := "52:55:00:d1:56:01"
+	primaryIp := "192.168.0.55"
+
+	setupBondWithDhcpTest(c, primaryMac, secondaryMac, primaryIp)
+
+	checkOvsBridge(c, primaryMac, primaryIp)
+}
+
+// setupDHCPServerContainer creates a dnsmaq container inside the vm that will interact with the veth pairs
+func setupDHCPServerContainer(c cluster.TestCluster, primaryMac, primaryIp string) {
+	m := c.Machines()[0]
+
+	genDHCPServerContainer(c,m)
+
+	wg := worker.NewWorkerGroup(context.Background(), 1)
+
+	dCmd := "podman run --rm dnsmaq --privileged --network ns:/var/run/netns/container dnsmasq"
+
+	dhcpContainer := func(ctx context.Context) error {
+		output, err := c.SSH(m, dCmd)
+		if err != nil {
+			return fmt.Errorf("failed to run %q: output: %q status: %q", dCmd, output, err)
+		}
+		return nil
+	}
+
+	if err := wg.Start(dhcpContainer); err != nil {
+		c.Fatal(wg.WaitError(err))
+	}
+}
+
+// make a podman container out of binaries on the host
+func genDHCPServerContainer(c cluster.TestCluster, m platform.Machine) {
+	//cmd := `tmpdir=$(mktemp -d); cd $tmpdir; echo -e "FROM scratch\nRUN dnf -y install systemd dnsmasq iproute iputils && dnf clean all && systemctl enable dnsmasq\nRUN echo -e "dhcp-range=192.168.0.50,192.168.0.60,12h\ndhcp-host=52:55:00:d1:56:00,192.168.0.55" > /etc/dnsmasq.d/dhcp && systemctl restart dnsmasq" > Dockerfile;
```

Comment:
The `systemctl restart dnsmasq` looks incorrect there. We want to build the image in this step, not run dnsmasq itself. Compare with https://github.com/coreos/fedora-coreos-config/blob/da83a0aa8d9cadad4ee132e7a7af1f9e88d08b59/tests/kola/chrony/dhcp-propagation#L37-L47.